### PR TITLE
Disable linters

### LINKED ISSUE #70: Feature Request: Enable Disabling Linters via Source
SCSS-lint has the ability to disable the linters within the source files

https://github.com/brigade/scss-lint#disabling-linters-via-source

This is unfortunately important for my project's workflow :)

### PR DESCRIPTION
**Pull request by DanPurdy**
Ongoing work for disabling linters via source.

closes #70 

Ongoing from the work of @donabrams in #402

Everyone feel free to contribute to this branch feature/disable-linters

### CODE REVIEW

**Comment by avinashci** [comment]
To start with the issue is in index.js file. In the sassLint.lintText method, following statements should be called after the ast is generated. 
    ruleToggles = getToggledRules(ast);
    isEnabledFilter = isResultEnabled(ruleToggles);
The above lines should move inside the try block. I made the changes locally and the tests for rules are passing. Though the cli tests are failing.

**Comment by DanPurdy** [comment]
Thanks @avinashci as mentioned there has been a lot of changes to sass-lint since this was last worked on including the block of code you highlighted the result you're seeing is just the result of the work being brought up to date with the current version and is unfortunately not finished.. 

Thanks for that catch though, feel free to open a PR into the disable linter branch to contribute that change. We'll be moving forward with updating this all  very soon

**Comment by FDiskas** [comment]
:telephone: must have

**Comment by DanPurdy** [comment]
Please do not add comments in PR's unless they add something to the discussion. Issues can be used to request features and github's reactions can be used to show your support for things.

**Comment by saptar** [comment]
submitted a pull request https://github.com/sasstools/sass-lint/pull/723 to incorporate changes for programatically enabling/disabling linting ; as a part of issue # 70.

**Comment by saptar** [comment]
Hi @DanPurdy ,
I have updated the PR https://github.com/sasstools/sass-lint/pull/723 .
Thanks!
-Saptarshi

**Comment by avinashci** [comment]
Looks like the merge issues are resolved and the changes have been integrated to the branch. Anymore help needed here?

**Comment by DanPurdy** [comment]
There are no tests for `.sass` syntax and I've not had time to go through it properly but I will.. Yeah the main thing holding it back at the moment is the sass tests

**Comment by saptar** [comment]
Hi @DanPurdy ,

I have created a PR https://github.com/sasstools/sass-lint/pull/743 to add test cases for .sass syntax.

Thanks!

**Comment by stevegibbings** [comment]
My team mates and I would like to contribute to this. How best can we help? I realise we are a little late to the party;-)

**Comment by DanPurdy** [comment]
Hi @stevegibbings, Thanks for the offer any help provided would be appreciated (personally struggling to find time at moment!) 

There's an issue in https://github.com/sasstools/sass-lint/pull/743 that would need looking at in order to make sure the `Sass` syntax is working correctly. It looks like an issue with Sass and CRLF line endings.

Alternatively throwing a few more test cases at both syntaxes with a view to just releasing this functionality for `scss` in the meantime would be appreciated. 

Basically any help or light that can be shed on this feature would be greatly appreciated.

**Comment by DanPurdy** [comment]
Rebased with develop to continue work on this. I intend this to arrive with the next version of sass-lint.

File: lib/ruleToggler.js [hunk 1]
```diff
@@ -118,10 +209,14 @@ module.exports.getToggledRules = function (ast) {
...
-  return x < x2 || (x === x2 && y < y2);
-};
-
+/**
+ * Filters our rule results by checking the lint result and it's line/column against our
+ * toggledRules object to see whether we should still be reporting this lint.
+ *
+ * @param {Object} toggledRules - The toggledRules object containing all of our rule enable/disable information
+ *
+ * @returns {Boolean} Whethere the current rule is disabled for this lint report
```
Comment (glen-84) [inline lib/ruleToggler.js]: Typo.

**Comment by paigr** [comment]
In regard to #743; I have submitted an [issue](https://github.com/tonyganch/gonzales-pe/issues/224) and [pull request](https://github.com/tonyganch/gonzales-pe/issues/225) to gonzales, with a potential fix for the CRLF bug. I will provide another update pending their review.

**Comment by DanPurdy** [comment]
@paigr Thanks for that, I'll ask the maintainer to take a look for you shortly, hopefully we can get this merged in soon. I'm just getting back into the swing of things after a break so would expect to see a bit more momentum shortly.

**Comment by DanPurdy** [comment]
Well this is promising. Great work everyone!

Now before I merge this in I would like to just throw this branch at some real world use cases and make sure it's working as expected. If anyone wants to help out and report their findings back here that would be awesome, otherwise this looks good for 1.10!

**Comment by macdonaldr93** [comment]
@DanPurdy we've been using this branch for a while now and haven't had any issues. I just synced Shopify's fork and we'll be testing this on our internal projects. Will let you know if there's anything that we notice.

**Comment by DanPurdy** [comment]
@macdonaldr93 That is great news, thanks for letting me know. Just out of curiosity can I get a bit of info off of you?

What line endings do you use? LF or CRLF?
Sass or scss?
Which platform are you running? OSX, Windows?

Thanks again 👍

**Comment by macdonaldr93** [comment]
@DanPurdy for sure.

- Line endings: LF
- scss
- OSX (I'm on 10.11.6, I know some of the others are on Sierra now)

**Comment by pselden** [comment]
I pulled this branch, npm linked it, than ran it on my current project and got a few errors that I wasn't getting before:

screen shot 2016-10-25 at 9 51 31 am
screen shot 2016-10-25 at 9 51 37 am

```scss

**Comment by bgriffith** [comment]
Hey @pselden  could you post a new issue with the code causing those errors? It looks like an issue with our AST (separate to this PR), thanks!

**Comment by pselden** [comment]
@bgriffith -- Sure thing.

**Comment by DanPurdy** [comment]
@pselden did you work out why it was failing for you or is it something we need to look into? Thanks

**Comment by pselden** [comment]
https://github.com/sasstools/sass-lint/pull/677#issuecomment-256599339 -- I did not, yet -- I was going to file a separate issue but it also failed on master, but passed when using the exact same version when installed directly from npm -- so I think it is probably just a build or install step i missed?

**Comment by pselden** [comment]
Here's what it was failing on though, if you were interested: https://gist.github.com/pselden/e58c7120069d961e52bf401fe8c69c29 (it's the interpolation of the class name when nested inside another one)

**Comment by DanPurdy** [comment]
Ah that's a bug with a previous version of our AST, shouldn't cause any issues now when we release the latest version of Sass-lint

**Comment by DanPurdy** [comment]
Will release with 1.10

**Comment by albell** [comment]
@DanPurdy Thank you! When should folks generally expect 1.10 to roll out?

**Comment by bgriffith** [comment]
@albell It's already out! 🎉
